File: members/U02CDR60PFS/kakao2021/합승_택시_요금.js
function solution(n, s, a, b, fares) {
    const INF = 10000000;
    let answer = INF;
    const floyd = Array.from(Array(n+1), () => Array(n+1).fill(INF)).map((e, i) => {e[i] = 0; return e});

    fares.forEach((e) => {
        floyd[e[0]][e[1]] = floyd[e[1]][e[0]] = e[2];
    })

    for(let k=1; k<=n; k++){
        for(let i=1; i<=n; i++){
            for(let j=1; j<=n; j++){
                if(floyd[i][k] + floyd[k][j] < floyd[i][j]){
                    floyd[i][j] = floyd[i][k] + floyd[k][j];
                }
            }
        }
    }
    
    for(let k=1; k<=n; k++){
        const dist = floyd[s][k] + floyd[k][a] + floyd[k][b];
        answer = dist > answer ? answer : dist;
    }
    
    return answer;
}

File: members/U02BQ6G1SQJ/programmers/problem/합승 택시 요금.py
# 소요 시간 : 17분
import math;

def printMatrix(nodeCnt : int, matrix : list) -> None:
    for r in range(1, nodeCnt + 1):
        print(matrix[r]);
    
    return None;

def solution(n : int, s : int, a : int, b : int, fares : list) -> int:
    (answer, adjMatrix) = (math.inf, [[math.inf for _ in range(n + 1)] for _ in range(n + 1)]);
    
    for (src, dst, fare) in fares:
        adjMatrix[src][dst] = fare;
        adjMatrix[dst][src] = fare;
    
    for passNode in range(1, n + 1):
        for startNode in range(1, n + 1):
            for endNode in range(1, n + 1):
                adjMatrix[startNode][endNode] = (0 if (startNode == endNode) else min(adjMatrix[startNode][endNode], adjMatrix[startNode][passNode] + adjMatrix[passNode][endNode]));
    
    # printMatrix(n, adjMatrix);
    
    for shareNode in range(1, n + 1):
        answer = min(answer, adjMatrix[s][shareNode] + adjMatrix[shareNode][a] + adjMatrix[shareNode][b]);
    
    return answer;
